doc/migration.md

```diff
@@ -0,0 +1,158 @@
+# Package changes
+
+Classes in the library have been re-organised to simplify the packaging.
+
+All classes are now in the `com.cloudant.sync` package:
+
+* All private API classes are in the `com.cloudant.sync.internal`
+  package. API users are discouraged from using these classes as fields,
```

Comment:
Is `discouraged` a strong enough word for this now that the segregation is so clear?
`should not` or maybe even `must not`
>API users **must not** directly use the classes in these packages as ... .

Maybe consider something along the lines of:
>Directly calling any classes or methods in these packages is unsupported.